### PR TITLE
Update C++ language extensions

### PR DESCRIPTION
**Pull request by matyas-streamhpc**

### CODE REVIEW

File: docs/reference/kernel_language.rst [hunk 1]
```diff
@@ -1327,8 +1357,7 @@ To read a high-resolution timer from the device, HIP provides the following buil
 
   .. code:: cpp
 
-    clock_t clock()
-    long long int clock64()
+    long long int wall_clock64()
```
Comment (TomSang) [inline docs/reference/kernel_language.rst]: The following should not be removed , 
__device__ long long int clock();
__device__ long long int clock64();

But we need note that clock() and clock64() do not work normally in Gfx11.

File: docs/reference/kernel_language.rst [hunk 2]
```diff
@@ -51,8 +51,7 @@ Supported ``__global__`` functions are:
   * Run on the device
   * Called (launched) from the host
 
-HIP ``__global__`` functions must have a ``void`` return type. The first parameter in a HIP ``__global__``
-function must have the type ``hipLaunchParm``. Refer to :ref:`kernel-launch-example` to see usage.
```
Comment (TomSang) [inline docs/reference/kernel_language.rst:55]: "hipLaunchParm lp" also need be removed from 

__global__ MyKernel(hipLaunchParm lp,...)

in sample code.

**Review by TomSang — CHANGES_REQUESTED** [review]

**Comment by TomSang** [comment]
I have a few comments.  Please update PR accordingly.

File: docs/reference/kernel_language.rst (hunk 1 above)
Comment (matyas-streamhpc) [reply docs/reference/kernel_language.rst]: Updated

File: docs/reference/kernel_language.rst (hunk 2 above)
Comment (matyas-streamhpc) [reply docs/reference/kernel_language.rst:55]: Removed

File: docs/reference/kernel_language.rst [hunk 3]
```diff
@@ -1327,16 +1357,16 @@ To read a high-resolution timer from the device, HIP provides the following buil
...
 
   The difference between the values that are returned represents the cycles used.
 
 * Returning the wall clock count at a constant frequency on the device:
 
   .. code:: cpp
 
-    long long int wall_clock64()
+    clock_t clock()
+    long long int clock64()
```
Comment (TomSang) [inline docs/reference/kernel_language.rst]: long long int clock()
long long int clock64()
Should be before line 1362

long long int wall_clock64()
should be after line 1367

**Review by TomSang — CHANGES_REQUESTED** [review]

**Review by TomSang — CHANGES_REQUESTED** [review]
"hipLaunchParm lp, " should be removed here,
// Example hipLaunchKernelGGL pseudocode:
  __global__ MyKernel(hipLaunchParm lp, float *A, float *B, float *C, size_t N)
  {
  ...
  }

**Comment by neon60** [comment]
@matyas-streamhpc rebased your branch and targeted docs/develop

File: docs/reference/kernel_language.rst (hunk 3 above)
Comment (matyas-streamhpc) [reply docs/reference/kernel_language.rst]: Fixed.

**Comment by matyas-streamhpc** [comment]
> "hipLaunchParm lp, " should be removed here, // Example hipLaunchKernelGGL pseudocode: **global** MyKernel(hipLaunchParm lp, float *A, float *B, float *C, size_t N) { ... }

It has been fixed in https://github.com/ROCm/HIP/pull/3488/commits/3be5ac382846443cbcb647734890b6932b649222
(former https://github.com/ROCm/HIP/pull/3488/commits/d518b3a9e4955074b1535a1f8f6b4ed4524bc794)

**Comment by matyas-streamhpc** [comment]
> @matyas-streamhpc rebased your branch and targeted docs/develop

Rebased again.

**Review by TomSang — APPROVED** [review]

**Pull request merged**